fs/copy.go

```diff
@@ -158,7 +158,9 @@ func copyDirectory(dst, src string, inodes map[uint64]string, o *copyDirOpts) er
 			}
 		default:
 			// TODO: Support pipes and sockets
-			return fmt.Errorf("unsupported mode %s: %w", fi.Mode(), err)
+			if err != nil {
+				return fmt.Errorf("unsupported mode %s: %w", fi.Mode(), err)
+			}
```

Comment:
This does not fix https://github.com/containerd/continuity/pull/185#discussion_r833676873 . It will just skip over this condition until it reaches `copyFileInfo` and errors there because the path does not exist.

Reply:
covered in:
- https://github.com/containerd/continuity/pull/197